service/history/transferQueueStandbyProcessor.go

```diff
@@ -222,6 +252,13 @@ func (t *transferQueueStandbyProcessorImpl) processCloseExecution(transferTask *
 	processTaskIfClosed := true
 	return t.processTransfer(processTaskIfClosed, transferTask, func(msBuilder *mutableStateBuilder) error {
 
+		ok, err := t.verifyVersion(transferTask.DomainID, msBuilder.GetCurrentVersion(), transferTask)
```

Comment:
Can this logic be put within base processTransfer function which already takes care of loading the mutable state?

Reply:
nop, some task requires checking whether a certain task, i.e. activity task is finished or not before checking the version, sometimes, i.e. close execution, do not require that.
By moving the checking within the function which load the mutable state, the logic will be nested